dashboard/src/shared/specs/mountWrapper.tsx

```diff
@@ -31,7 +36,7 @@ export const defaultStore = mockStore(initialState);
 // getStore returns a store initialised with a merge of
 // the initial state with any passed extra state.
 export const getStore = (extraState: object) => {
-  const state = { ...initialState };
+  const state = cloneDeep(initialState);
```

Comment:
Ouch - did you find that by tests writing state then seen in other tests?

Reply:
Yep, `{...object}` only performs a shallow copy, keeping references to inner objects:

```
> a = {a: 1, b: {c: 2}}
{ a: 1, b: { c: 2 } }
> b = {...a}
{ a: 1, b: { c: 2 } }
> b.b.c = 3
3
> a
{ a: 1, b: { c: 3 } }
```

Reply:
Yep yep, I realise, but also realised that I wrote that code the other week and was saying "ouch" because you may have had to find that bug the hard way (ie. tests failing because of contaminated test data)... hopefully you just noticed it while passing by and didn't suffer as a result :)

Reply:
sorry, I found this with weird test results :P but it was easy to spot